libbeat/publisher/queue/diskqueue/state_file.go

```diff
@@ -0,0 +1,101 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package diskqueue
+
+import (
+	"bufio"
+	"encoding/binary"
+	"fmt"
+	"os"
+)
+
+// Given an open file handle to the queue state, decode the current position
+// and return the result if successful, otherwise an error.
+func queuePositionFromHandle(
+	file *os.File,
+) (queuePosition, error) {
+	_, err := file.Seek(0, 0)
```

Comment:
Why is seeking necessary here? The caller is passing a newly opened file. Is this some kind of fool proofing? :D

Reply:
Well, the _current_ caller is passing a newly opened file, but this helper works on any handle, even one that's been read from before. (This isn't strictly necessary since the current design only caches writable handles, but I like to be cautious.)